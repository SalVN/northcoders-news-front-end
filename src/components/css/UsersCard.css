.users-list-image {
    width: 36px;
    height: 36px;
    margin-right: 20px;
}

.users-list-username-text {
    color: #c1002e;
    font-weight: 600;
    font-size: 20px;
    margin-top: -10px;
    margin-bottom: 5px;
}

.users-box-style {
    margin-left: 20px;
    margin-right: 20px;
}
