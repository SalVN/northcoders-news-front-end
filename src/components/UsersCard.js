import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Link } from 'react-router-dom';

import './css/UsersCard.css';

const UsersCard = function (props) {
    return (
        <div className='card users-box-style' key={props.user.username}>
                <div className='column is-two-thirds'>
                    <Link to={`/users/${props.user.username}`}>
                        <div>
                            <img className='users-list-image' src={`${props.user.avatar_url}`} />
                            <span className='users-list-username-text'>
                                {props.user.username}
                            </span>
                        </div>
                    </Link>
                </div>
            </div>
    );
};

UsersCard.propTypes = {
    user: PropTypes.object.isRequired,
};

export default UsersCard;