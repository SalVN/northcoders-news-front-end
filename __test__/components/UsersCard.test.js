import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import UsersCard from '../../src/components/UsersCard';

describe('UsersCard', () => {
    const user = {
        _id: '59b1b18b327cce1fb043bdb1',
        username: 'northcoder',
        name: 'Awesome Northcoder',
        avatar_url: 'https://avatars3.githubusercontent.com/u/6791502?v=3&s=200',
        __v: 0
    };
    it('is a function', () => {
        expect(typeof UsersCard).toEqual('function');
    });

    it('renders', () => {
        const enzymeWrapper = shallow(<UsersCard
            user={user}
        />);
        expect(enzymeWrapper.children().length).toEqual(1);
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <UsersCard
                    user={user}
                />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});