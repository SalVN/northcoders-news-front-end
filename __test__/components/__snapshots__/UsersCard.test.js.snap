// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersCard renders correctly 1`] = `
<div
  className="card users-box-style"
>
  <div
    className="column is-two-thirds"
  >
    <a
      href="/users/northcoder"
      onClick={[Function]}
    >
      <div>
        <img
          className="users-list-image"
          src="https://avatars3.githubusercontent.com/u/6791502?v=3&s=200"
        />
        <span
          className="users-list-username-text"
        >
          northcoder
        </span>
      </div>
    </a>
  </div>
</div>
`;
